cmd/mtaBuild.go

```diff
@@ -0,0 +1,411 @@
+	"github.com/SAP/jenkins-library/pkg/log"
+	"github.com/SAP/jenkins-library/pkg/piperutils"
+	"github.com/SAP/jenkins-library/pkg/telemetry"
+	"gopkg.in/yaml.v2"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+	"text/template"
+	"time"
+)
+
+const templateMtaYml = `_schema-version: "2.0.0"
+ID: "{{.ID}}"
+version: {{.Version}}
+
+parameters:
+  hcp-deployer-version: "1.0.0"
+
+modules:
+  - name: {{.Name}}
+    type: html5
+    path: .
+    parameters:
+       version: {{.Version}}-${timestamp}
+    build-parameters:
+      builder: grunt
+      build-result: dist`
+
+// MTABuildTarget ...
+type MTABuildTarget int
+
+const (
+	// NEO ...
+	NEO MTABuildTarget = iota
+	// CF ...
+	CF MTABuildTarget = iota
+	//XSA ...
+	XSA MTABuildTarget = iota
+)
+
+// ValueOfBuildTarget ...
+func ValueOfBuildTarget(str string) (MTABuildTarget, error) {
+	switch str {
+	case "NEO":
+		return NEO, nil
+	case "CF":
+		return CF, nil
+	case "XSA":
+		return XSA, nil
+	default:
+		return -1, fmt.Errorf("Unknown BuildTarget: '%s'", str)
+	}
+}
+
+// String ...
+func (m MTABuildTarget) String() string {
+	return [...]string{
+		"NEO",
+		"CF",
+		"XSA",
+	}[m]
+}
+
+func mtaBuild(config mtaBuildOptions, telemetryData *telemetry.CustomData, commonPipelineEnvironment *mtaBuildCommonPipelineEnvironment) {
+	log.Entry().Info("Launching mta build")
+	err := runMtaBuild(config, commonPipelineEnvironment, &command.Command{})
+	if err != nil {
+		log.Entry().
+			WithError(err).
+			Fatal("failed to execute mta build")
+	}
+
+}
+
+func runMtaBuild(config mtaBuildOptions, commonPipelineEnvironment *mtaBuildCommonPipelineEnvironment,
+	e envExecRunner) error {
+
+	e.Stdout(os.Stderr) // keep stdout clear.
+	e.Stderr(os.Stderr)
+
+	if len(config.ProjectSettingsFile) > 0 {
+
+		projectSettingsFileDest, err := getProjectSettingsFileDest()
+		if err != nil {
+			return err
+		}
+
+		if err = materialize(config.ProjectSettingsFile, projectSettingsFileDest); err != nil {
+			return err
+		}
+
+	} else {
+
+		log.Entry().Debugf("Project settings file not provided via configuation.")
+	}
+
+	if len(config.GlobalSettingsFile) > 0 {
+
+		globalSettingsFileDest, err := getGlobalSettingsFileDest()
+		if err != nil {
+			return err
+		}
+
+		if err = materialize(config.GlobalSettingsFile, globalSettingsFileDest); err != nil {
+			return err
+		}
+	} else {
+
+		log.Entry().Debugf("Global settings file not provided via configuation.")
+	}
+
+	if len(config.DefaultNpmRegistry) > 0 {
+		log.Entry().Debugf("Setting default npm registry to \"%s\"", config.DefaultNpmRegistry)
+		if err := e.RunExecutable("npm", "config", "set", "registry", config.DefaultNpmRegistry); err != nil {
+			return err
+		}
+	} else {
+		log.Entry().Debugf("No default npm registry provided via configuration. Leaving npm config untouched.")
+	}
+
+	mtaYamlFile := "mta.yaml"
+	mtaYamlFileExists, err := piperutils.FileExists(mtaYamlFile)
+
+	if err != nil {
+		return err
+	}
+
+	if !mtaYamlFileExists {
```

Comment:
Move `mta.yml` creation to dedicated package.

Reply:
We can do this. Is there any need for re-use? Otherwise having a better structure with facoring this out into a `func` might be sufficient.